operator/v1/types_csi_cluster_driver.go

```diff
@@ -40,6 +40,18 @@ type ClusterCSIDriver struct {
 // CSIDriverName is the name of the CSI driver
 type CSIDriverName string
 
+// +kubebuilder:validation:Pattern=`^(Managed|Unmanaged|Removed)$`
```

Comment:
You can use an enum tag for this. As this is adding to an existing API, you also need to account for an empty value
```suggestion
// +kubebuilder:validation:Enum="";Managed;Unmanaged;Removed
```

Reply:
putting my k8s api hat - it is pretty common to use pointer types for optional fields which default to a certain behaviour. I do not want to include empty string in allowed values, unless there is a good reason for doing so.